packages/decoder/lib/decoders.ts

```diff
@@ -1707,7 +1709,10 @@ export class ContractInstanceDecoder {
         break;
       case "mapping":
         let keyType = parentType.keyType;
-        if (keyType.typeClass === "enum") {
+        if (
+          keyType.typeClass === "enum" ||
+          keyType.typeClass === "userDefinedValueType"
+        ) {
```

Comment:
Is this correct? enum and UDVT are different.

Reply:
Yes, but both need to be covered here.  Notice what this if block does -- it fetches the full type information for the type.  We need that for both enums and for UDVTs, because we need the full type information to perform wrapping.  For enums we need to know the enum's overall range, and we need to know the names of its options to allow people to use them as input.  For UDVTs we need to know the underlying type because, well, everything is based on that!

Of course, there are two other types of user-defined types; however for contracts we don't need the full type information because we already know everything relevant for wrapping purposes, and as for structs, well, those aren't elementary types and so aren't relevant here.

Now you might say, isn't handling this here a bit awkward?  To which I say, yes!  Arguably, this code belongs inside `wrapElementaryValue` itself.  However, I didn't need to see a need to change it in this PR; especially as, once the encoder work is merged, this code will be eliminated.  So I'm fine with leaving it this way for now and just adding the extra typeclass to the condition.

Reply:
(Alternately, we could eliminate the check and just retrieve the full type information no matter what!  I just did it the same way it already was because like I said I didn't see a need to redo this.)

Reply:
Thanks for the explanation!